marlowe-runtime/doc/open-roles.md

```diff
@@ -0,0 +1,652 @@
+
+```yaml
+roles:
+  Thread:
+    script: ThreadRole
+  Buyer:
+    script: OpenRole
+  Seller: addr_test1vq9prvx8ufwutkwxx9cmmuuajaqmjqwujqlp9d8pvg6gupczgtm9j
+```
+
+### Pre-minted open roles
+
+If role tokens have already been minted and are held in the wallet of the contract's creator, then specify the policy ID, the name of the thread token, and the names of the open-role tokens. In the example below, the creation transaction will send the thread role token `Thread` to the Marlowe contract and the open role `Buyer` to the open-roles validator. Any standard role tokens will have to be distributed to parties manually.
+
+```yaml
+roles:
+  script: OpenRole
+  policyId: ac5591a09baca1e04b5b2f086210cd1a9451032e02c1a327d13ee767
+  threadRoleName: Thread
+  openRoleNames:
+    - Buyer
+```
+
+### Distribution of open-role tokens
+
+The open-role validator authorizes the distribution of the open-role token that it holds to anyone who performs the contract's first action that takes the open role as input. After they have received the open-role token, they can use it as a role token for any future inputs to the contract or for any withdrawals from the role-payout validator address.
+
+Furthermore, the open-role validator will only validate if the Marlowe contract holds the thread token specified when the open-role validator is created. This ties the open role to a particular Marlowe contract.
+
+
+### Security restriction for open roles
+
+Marlowe's prevention of double-satisfaction attacks requires that no external payments be made from a Marlowe contract if another Plutus script runs in the transaction. Thus, if an open-role is distributed in a transaction, the transaction cannot do `Pay` to parties or implicit payments upon `Close`. Typically, the distribution of an open-role in a Marlowe contract will be followed by a `Notify TrueObs` case so that further execution of the contract does not proceed in that transactions. External payments can be made in subsequent transactions.
+
+
+## Example
+
+The open-roles validator lets *anyone with a Cardano address* make a deposit or choice input for a contract that uses open roles. A simple use case for open roles is when a seller might offer a token for sale to anyone who pays the sales price.
+
+1. The seller creates a contract that offers one `BearGarden` token for sale at the price of 8 ada.
+2. The seller deposits the `BearGarden` token.
+3. Anyone can join the contract by depositing the sale price of 8 ada into the contract. In that transaction, the open-role validator will pay them the `Seller` token so that they are now a party to the contract.
```

Comment:
```suggestion
3. Anyone can join the contract by depositing the sale price of 8 ada into the contract. In that transaction, the open-role validator will pay them the `Buyer` token so that they are now a party to the contract.
```
The buyer of BearGuarden token probably gets the Buyer token, right?

Reply:
Thanks so much! Fixed.